src/ai/ask.rs

```diff
@@ -0,0 +1,116 @@
+use async_openai::{
+    config::OpenAIConfig,
+    types::{
+        ChatCompletionRequestMessage, ChatCompletionRequestMessageArgs,
+        CreateChatCompletionRequest, CreateChatCompletionRequestArgs, CreateModerationRequestArgs,
+        Role,
+    },
+    Client,
+};
+use futures_util::{stream::FuturesUnordered, TryStreamExt};
+use serde::Serialize;
+
+use crate::{
+    ai::{
+        constants::{ASK_SYSTEM_MESSAGE, ASK_USER_MESSAGE, MODEL},
+        embeddings::get_related_docs,
+        error::AIError,
+        helpers::{cap_messages, into_user_messages, sanitize_messages},
+    },
+    db::SupaPool,
+};
+
+#[derive(Serialize)]
+pub struct RefDoc {
+    pub slug: String,
+    pub title: String,
+}
+
+pub struct AskRequest {
+    pub req: CreateChatCompletionRequest,
+    pub refs: Vec<RefDoc>,
+}
+
+pub async fn prepare_ask_req(
+    client: &Client<OpenAIConfig>,
+    pool: &SupaPool,
+    messages: Vec<ChatCompletionRequestMessage>,
+) -> Result<AskRequest, AIError> {
+    let open_ai_messages = sanitize_messages(messages);
```

Comment:
We never use these role assistant messages, is this step necessary?

Reply:
That's to be safe. We _don't_ need it right now but we might. And this protects against injecting system messages.